### PR TITLE
New Mod: Martial Mastery

### PR DESCRIPTION
**Pull request by John-Candlebury**
#### Summary
Mods "Create the Martial Mastery mod, which allows you to develop your own martial art using a level up system "

#### Purpose of change
The spell system and related enchantment content currently covers the make your own wizard part of the fantasy RPG trio, but there's very little for players that would like to play warrior or rogue archetypes.  This mod seeks to resolve that by letting players build a custom fighting style by selecting different perk locked abilities. 

The mod seeks to respect the main game's outlook on the melee combat system. The player imput's for combat will be limited to movement, waiting and attacking,  but complex actions can be set up by different rhythms in how these actions are chosen. 

#### Describe the solution
This mod overwrites the default martial art and plugs a wide variety of perks that unlock new techniques and buffs as in bombastic perks, you earn XP by defeating enemies and can level up to unlock new abilities.

This first release focuses on the attack and movement actions, and principally adds techniques for knives, swords and spears. It includes 20+ new perks that have some basic synergies between them.

#### Describe alternatives you've considered

A larger release but I think this is a good starting point atm.

#### Testing
A lot of combat.

### CODE REVIEW

File: data/mods/Perk_melee/techniques/techniques.json [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    "crit_ok": true,
+    "dodge_counter": true,
+    "weighting": 100,
+    "mult_bonuses": [
+      { "stat": "movecost", "scale": 0.1 },
+      { "stat": "damage", "type": "bash", "scale": 0.02 },
+      { "stat": "damage", "type": "cut", "scale": 0.2 },
+      { "stat": "damage", "type": "stab", "scale": 0.2 }
+    ],
+    "messages": [ "You slash at the %s as you dodge", "<npcname> shaleshes at the %s as they dodge" ],
```
Comment (John-Candlebury) [inline data/mods/Perk_melee/techniques/techniques.json:213]:
```suggestion
    "messages": [ "You slash at the %s as you dodge", "<npcname> slashes at %s as they dodge" ],
```

**Comment by Aimless-dude** [comment]
If I can make a suggestion, it might be neat if you could learn techniques to use in your custom style by knowing techniques from other fighting styles. Like if you know Boxing, being able to incorporate a Cross Counter into the custom style. If that's, at all possible.

**Comment by John-Candlebury** [comment]
No Im not interested in doing that. A lot of specific attacks only make sense in their own martial arts, and I seeking to do something more generic.

**Comment by ashGlaw** [comment]
You have my attention! Very exciting

**Comment by John-Candlebury** [comment]
> Is XP restrictable to melee kills only? "You shoot your way into punching better" doesn't make sense 😛

Unfortunately it isn't. 
I think its not overtly terrible because I hope to include upgrades for both bows and crossbows in due time (just as theres a sizeable buff to throwing in one of the perks here)

**Comment by Cormac1991** [comment]
adding complexity to a system most people dont understand.

**Pull request merged**